### PR TITLE
Fix some memory leaks

### PR DESCRIPTION
**Pull request by IvanNardi**

### CODE REVIEW

**Comment by IvanNardi** [comment]
Some fixes seem pretty easy, but other ones are not so obvious...

File: picoquictest/tls_api_test.c [hunk 1]
```diff
@@ -7155,6 +7155,9 @@ int bad_cnxid_test()
             ret = -1;
         }
         else {
+            for (size_t i = 0; i < test_ctx->nb_test_streams; i++) {
+                test_api_delete_test_stream(&test_ctx->test_stream[i]);
+            }
```
Comment (IvanNardi) [inline picoquictest/tls_api_test.c:7160]: These frees are needed, but they seems out of place here...

File: picoquictest/tls_api_test.c [hunk 2]
```diff
@@ -8616,6 +8619,9 @@ int ddos_amplification_test_one(int use_0rtt)
             picoquic_delete_cnx(test_ctx->cnx_server);
             /* Delete the old client connection */
             picoquic_delete_cnx(test_ctx->cnx_client);
+            for (size_t i = 0; i < test_ctx->nb_test_streams; i++) {
+                 test_api_delete_test_stream(&test_ctx->test_stream[i]);
+            }
```
Comment (IvanNardi) [inline picoquictest/tls_api_test.c:8624]: Same as above

File: picoquictest/tls_api_test.c (hunk 1 above)
Comment (huitema) [reply picoquictest/tls_api_test.c:7160]: Yes. They belong in the deletion of the test context.

File: picoquictest/tls_api_test.c (hunk 2 above)
Comment (huitema) [reply picoquictest/tls_api_test.c:8624]: Yes. And that's hiding the missing code in the test context deletion.

**Review by huitema — COMMENTED** [review]
Looks good. I was going though some of that in PR #1139. I will check you PR first.

**Pull request merged**